### PR TITLE
Allow adding more claims to OIDC JWT authentication token

### PR DESCRIPTION
**Pull request by sberyozkin**
Replaces #34686.

@lordvlad, can you please check this PR ?

I thought first I'd start cherry-picking from your PR but then decided not to since having a Map as we agreed in #34686 makes code changes different. I thought it would be simpler to recreate a PR due to the merge issues in your original PR

### CODE REVIEW

**Comment by lordvlad** [comment]
This looks really good. Much appreciated

File: extensions/oidc-common/runtime/src/main/java/io/quarkus/oidc/common/runtime/OidcCommonUtils.java [hunk 1]
```diff
@@ -367,6 +368,11 @@ public static String signJwtWithKey(OidcCommonConfig oidcConfig, String tokenReq
         }
     }
 
+    @SuppressWarnings({ "unchecked", "rawtypes" })
```
Comment (lordvlad) [inline extensions/oidc-common/runtime/src/main/java/io/quarkus/oidc/common/runtime/OidcCommonUtils.java:371]: This method is only introduced so you don't have to annotate signJwtWithKey with SuppressWarnings, right?

**Review by lordvlad — APPROVED** [review]

File: extensions/oidc-common/runtime/src/main/java/io/quarkus/oidc/common/runtime/OidcCommonUtils.java (hunk 1 above)
Comment (sberyozkin) [reply extensions/oidc-common/runtime/src/main/java/io/quarkus/oidc/common/runtime/OidcCommonUtils.java:371]: @lordvlad Yes, it is guaranteed Quarkus Config will init an empty or non empty Map with string value. When dealing directly with the JWT build api, Map can contains complex claims like arrays or objects

**Comment by sberyozkin** [comment]
Thanks @lordvlad for your original PR and the discussion which has led us to introducing a map, users should now be able to add whatever simple custom string claims they need. Support for complex claims, non String types can be reviewed as needed

**Comment by sberyozkin** [comment]
@geoand @gastaldi Can you please have a quick review ? @lordvlad who created the original PR has approved so it should be good to go

**Review by gastaldi — APPROVED** [review]

**Comment by lordvlad** [comment]
@sberyozkin quick question, to understand the release process: why was this not included in 3.2.2.Final? Cause the scope for 3.2.2 was already set? Will it be included in 3.2.3?

**Comment by gastaldi** [comment]
It will be included in 3.3.0.CR1, due in August 9th, see https://github.com/quarkusio/quarkus/wiki/Release-Planning

**Comment by sberyozkin** [comment]
@lordvlad Sometimes we can request backports for bug fixes - but this one can be considered a new feature, which is why it is planned for 3.3.0.CR1, I hope you'll have no problems migrating

**Comment by lordvlad** [comment]
👍

**Pull request merged**